src/features/filter.ts

```diff
@@ -1,12 +1,38 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
+import { RootState } from '../app/store';
 
 const initialState = {
   query: '',
-  status: 'all',
+  status: Status.All,
 };
 
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setStatusFilter: (state, action: PayloadAction<Status>) => {
+      return {
+        ...state,
+        status: action.payload,
+      };
+    },
+    setSearchQuery: (state, action: PayloadAction<string>) => {
+      return {
+        ...state,
+        query: action.payload,
+      };
+    },
+    resetFilters: () => {
+      return initialState;
```

Comment:
The `resetFilters` reducer directly returns the `initialState`, which is fine. However, if the `initialState` is ever mutated, it could lead to unexpected behavior. Consider returning a new object with the same properties as `initialState` to ensure immutability.